openapi-spec/swagger.yaml

```diff
@@ -0,0 +1,1472 @@
+        serial_number:
+          type: string
+          example: '0002004355'
+        location:
+          type: string
+        created_at:
+          type: string
+        updated_at:
+          type: string
+        sync_status:
+              type: string
+              enum:
+                - SYNCED
+                - SYNCING
+        total_capacity:
+          type: integer
+          format: int64
+        used_capacity:
+          type: integer
+          format: int64
+        free_capacity:
+          type: integer
+          format: int64
+    StorageAccessInfoResponse:
+      type: object
+      properties:
+        id:
+          type: string
+        username:
+          type: string
+          example: admin
+        port:
+          type: string
+          example: '1234'
+        vendor:
+          type: string
+          example: dellemc
+        model:
+          type: string
+          example: vmax
+        extra_attributes:
+          type: object
+          additionalProperties:
+            type: string
+          example:
+            array_id: string
+
+
+    StoragePoolSpec:
+      description: >-
+        A storage pool is disocovered and updated by task manager Each pool can be
+        regarded as a physical storage pool or a virtual storage pool. It is a
+        logical and atomic pool and can be abstracted from any storage platform.
+      allOf:
+        - $ref: '#/components/schemas/BaseModel'
+        - required:
+            - id
+            - name
+          type: object
+          properties:
+            name:
+              type: string
+            id:
+              type: string
+              readOnly: true
+              example: 084bf71e-a102-11e7-88a8-e31fe6d52248
+            native_storage_pool_id:
+              type: string
+              readOnly: true
+              example: SRP_1
+            storage_type:
+              type: string
+              enum:
+                - block
+                - file
+            description:
+              type: string
+            status:
+              type: string
+              enum:
+                - normal
+                - offline
+                - abnormal
+            total_capacity:
+              type: integer
+              format: int64
+            used_capacity:
+              type: integer
+              format: int64
+            free_capacity:
+              type: integer
+              format: int64
+    VolumeRespSpec:
+      description:
+        Volume is an device created by storage service, it can be attached to
+        physical machine or virtual machine instance.
+      allOf:
+        - $ref: '#/components/schemas/BaseModel'
+        - type: object
+          properties:
+            name:
+              type: string
+            description:
+              type: string
+            status:
+              type: string
+              enum:
+                - normal
+                - offline
+                - abnormal
```

Comment:
I seems the return of status is available and error.

Reply:
done